operator/pkg/helmreconciler/reconciler.go

```diff
@@ -188,14 +197,86 @@ func (h *HelmReconciler) processRecursive(manifests name.ManifestMap) *v1alpha1.
 	}
 	wg.Wait()
 
+	ostatus := overallStatus(componentStatus)
+	if ostatus == v1alpha1.InstallStatus_HEALTHY {
+		if err := h.checkResourceStatus(&componentStatus, &ostatus); err != nil {
```

Comment:
Why do we need to do another wait here, when in `ApplyManifest` we can wait already?

Reply:
that's for the case when user just update with running kubectl apply -f newcr

Reply:
ah actually now the new code would handles that case using the same flow though.. let me double check with Martin

Reply:
correct, this code is not required.

Reply:
removed